src/lib/molecules/AdvancedSearch/index.js

```diff
@@ -43,10 +44,14 @@ function AdvancedSearch({
     const searchRef = useRef(null);
     const [popoverOpen, setPopoverOpen] = useState(false);
     const [initialDataState, setInitialDataState] = useState([]);
-
     const [searchValue, setSearchValue] = useState(null);
+    const [isOpenControlled, setIsOpenControlled] = useState(false);
```

Comment:
Is controlled should be just a memoized variable as the user passes the `isOpen` prop which means the component will be controlled by the user. Why do you use here state? It will make an extra render after the set.

Reply:
```javascript
const isOpenControlled = 'isOpen' in props;
```